tests/Unit/API/Site/Controllers/MerchantCenter/RequestReviewControllerTest.php

```diff
@@ -110,12 +113,188 @@ public function test_merged_response() {
+					                 ]
+				                 ],
+				                 'programTypeB' => [
+					                 'status' => 200,
+					                 'data'   => [
+						                 "globalState"    => "ENABLED",
+						                 'regionStatuses' => [
+							                 [
+								                 'reviewEligibilityStatus' => 'INELIGIBLE',
+								                 'eligibilityStatus'       => RequestReviewStatuses::APPROVED,
+							                 ]
+						                 ]
+					                 ]
+				                 ],
+				                 'programTypeC' => [
+					                 'status' => 200,
+					                 'data'   => [
+						                 "globalState" => "UNKNOWN"
+					                 ]
+				                 ],
+			                 ]
+		                 );
+
+		$response = $this->do_request( self::ROUTE_GET_REQUEST );
+		$this->assertEquals( 200, $response->get_status() );
+		$this->assertEquals( [
+			'status'   => RequestReviewStatuses::APPROVED,
+			'issues'   => [],
+			'cooldown' => 0
+		], $response->get_data() );
+	}
+
+	public function test_unexpected_status_response() {
+
+		$this->middleware->expects( $this->once() )
+		                 ->method( 'get_account_review_status' )
+		                 ->willReturn(
+			                 [
+				                 'programTypeA' => [
+					                 'status' => 200,
+					                 'data'   => [
+						                 "globalState"    => "ENABLED",
+						                 'regionStatuses' => [
+							                 [
+								                 'reviewEligibilityStatus' => 'UNKNOWN',
+								                 'eligibilityStatus'       => 'UNKNOWN',
+							                 ],
+							                 [
+								                 'reviewEligibilityStatus' => 'ELIGIBLE',
+								                 'eligibilityStatus'       => RequestReviewStatuses::DISAPPROVED,
+								                 'reviewIssues'            => [ 'one' ]
+							                 ],
+							                 [
+								                 'reviewEligibilityStatus' => 'WEIRD',
+								                 'eligibilityStatus'       => 'WEIRD',
+							                 ],
+						                 ]
+					                 ]
+				                 ]
+			                 ]
+		                 );
+
+		$response = $this->do_request( self::ROUTE_GET_REQUEST );
+		$this->assertEquals( 200, $response->get_status() );
+		$this->assertEquals( [
+			'status'   => RequestReviewStatuses::DISAPPROVED,
+			'issues'   => [ 'one' ],
+			'cooldown' => 0
+		], $response->get_data() );
+	}
+
+	public function test_unset_region_statuses() {
+
+		$this->middleware->expects( $this->once() )
+		                 ->method( 'get_account_review_status' )
+		                 ->willReturn(
+			                 [
+				                 'programTypeA' => [
+					                 'status' => 200,
+					                 'data'   => [
+						                 "globalState"    => "TEST",
```

Comment:
💅

I think the globalState should be a valid state for example 'RequestReviewStatuses::ENABLE' or 'RequestReviewStatuses::NO_OFFERS' to cover the case of a valid globalState without region but as you mentioned in a previous comment, that will depend on:

>Thats a fair question, I wanted to avoid inconsistencies. But your'e right that maybe it's better to allow processing of other statuses.

Reply:
Thats handled in `test_unexpected_state_and_good_states_response` test case